### DISCUSSION TITLE
Arquitetura do projeto de segurança em APIs.

### DISCUSSION THREAD

**Discussion by LindomarSo** [balta-io/forum#402] [Tira Dúvidas] [answered]
Balta, tudo tranquilo? 

Gostei muito da arquitetura utilizada no curso por isso estou tentando me aprofundar mais. Nesse sentido essa abordagem tem alguma relação com a arquitetura vertical slice?

Outra coisa vc conseguiria me dar um exemplo de como ficaria a organização de pastas se eu tivesse um projeto com:

    - Autenticação 
    - Alunos 
    - Cursos

**Comment by andrebaltieri** [comment]
Bom dia, @LindomarSo, como vai?
Chegou a dar uma olhada nesse curso: https://balta.io/cursos/seguranca-apis-aspnet-jwt-bearer

Aí talvez tenha o exemplo que procura.

**Reply by LindomarSo** [reply]
Bom dia, Balta, Obrigado por responder!

Desculpa se a minha pergunta parece muito obvia, mas realmente eu fiquei com algumas dúvidas, em todo caso se tiver algum livro ou algum projeto em que eu posso dar uma olhada já resolve também.

Eu assistir, o curso e algumas aulas vi mais de uma vez também. No curso, vc fala que vai utilizar uma arquitetura mais vertical, por isso perguntei se tem alguma relação com a arquitetura vertical slice, lembrando que eu não tenho conhecimento sobre essa abordagem.

No curso não fica muito claro alguns pontos, por exemplo, se dentro do repositório do contexto eu posso fazer consultas a outras tabelas  onde eu vou pegar informações para a criação de um objeto, ou se nesse caso eu precisaria de um instância de um outro repositório.

Vou começar um projeto pessoal e gostaria de utilizar essa abordagem, mas não queira fazer errado. Desde já agradeço qualquer ajuda nesse sentido.

**Reply by andrebaltieri** [reply, marked as answer]
Bom dia, @LindomarSo, como vai?

Não existe pergunta óbvia, pode mandar...

"dentro do repositório do contexto eu posso fazer consultas a outras tabelas  onde eu vou pegar informações para a criação de um objeto"
- Sim, no repositório você tem acesso ao banco (Aplicação ou do contexto). Agora precisa entender pq está precisando de dados de outro contexto.

Sugiro rever os cursos da carreira de arquitetura, lá tem mais detalhes sobre DDD, Testes, Contextos e afins.
